mypy/test/testargs.py

```diff
@@ -17,3 +22,47 @@ def test_coherence(self) -> None:
         # FIX: test this too. Requires changing working dir to avoid finding 'setup.cfg'
         options.config_file = parsed_options.config_file
         assert_equal(options, parsed_options)
+
+    def test_executable_inference(self) -> None:
+        """Test the --python-executable flag with --python-version"""
+        sys_ver_str = '.'.join(map(str, sys.version_info[:2]))
+
+        base = ['file.py']  # dummy file
+
+        # test inference given one (infer the other)
+        matching_version = base + ['--python-version={}'.format(sys_ver_str)]
+        _, options = process_options(matching_version)
+        assert options.python_version == sys.version_info[:2]
+        assert options.python_executable == sys.executable
+
+        matching_version = base + ['--python-executable={}'.format(sys.executable)]
+        _, options = process_options(matching_version)
+        assert options.python_version == sys.version_info[:2]
+        assert options.python_executable == sys.executable
+
+        # test inference given both
+        matching_version = base + ['--python-version={}'.format(sys_ver_str),
+                                   '--python-executable={}'.format(sys.executable)]
+        _, options = process_options(matching_version)
+        assert options.python_version == sys.version_info[:2]
+        assert options.python_executable == sys.executable
+
+        # test that we error if the version mismatch
+        # argparse sys.exits on a parser.error, we need to check the raw inference function
+        options = Options()
+
+        special_opts = argparse.Namespace()
+        special_opts.python_executable = sys.executable
+        special_opts.python_version = (2, 10)  # obviously wrong
+        special_opts.no_executable = None
+        with pytest.raises(PythonExecutableInferenceError) as e:
+            options = infer_python_version_and_executable(options, special_opts)
+        assert str(e.value) == 'Python version (2, 10) did not match executable {}, got' \
+                               ' version {}.'.format(sys.executable, str(sys.version_info[:2]))
```

Comment:
nit: No need for the `str` here

Reply:
It is indeed needed. `type(e.value)` is `mypy.main.PythonExecutableInferenceError`.

Reply:
I meant the second `str` on `sys.version_info`